### PR TITLE
Ln/add map set func

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

File: lib/generics.go [hunk 1]
```diff
@@ -55,3 +55,19 @@ func (set *Set[T]) ToSlice() []T {
 	}
 	return results
 }
+
+func MapSet[T comparable, K any](set *Set[T], mapFunc func(elem T) (K, error)) ([]K, error) {
+	var results []K
+	err := set.ForEach(func(elem T) error {
+		mappedResult, innerErr := mapFunc(elem)
+		if innerErr != nil {
+			return innerErr
```
Comment (mattfoley8) [inline lib/generics.go:64]: I assume this works, but I'm not sure I follow how. We will end up with a mappedResult that is less than or equal to the length of the set. Less than if there are any errors. Equal if there aren't any errors. And we will end up with a map of returned errors or nils. How does the outer error know to error if e.g. `Any?([nil, nil, error, nil, nil])`.

Comment (lazynina) [reply lib/generics.go:64]: If any element returns an error from the map function, the whole thing errors. I can add a test for this

Comment (mattfoley8) [reply lib/generics.go:64]: Makes sense. +1 to a test. Would convince me definitively.

File: lib/db_utils.go [hunk 2]
```diff
@@ -8772,15 +8766,9 @@ func DBGetPostAssociationsByAttributes(
...
-		associationEntry, innerErr := DBGetPostAssociationByID(handle, snap, &associationID)
-		if innerErr != nil {
-			return innerErr
-		}
-		associationEntries = append(associationEntries, associationEntry)
-		return nil
+		return DBGetPostAssociationByID(handle, snap, &associationID)
 	})
 	if err != nil {
 		return nil, nil, errors.Wrapf(err, "DBGetUserAssociationsByAttributes: problem retrieving association entry by ID: ")
```
Comment (mattfoley8) [inline lib/db_utils.go:8774]: Unrelated to your changes but this should say DBGetPostAssociationsByAttributes…

**Review by mattfoley8 — APPROVED** [review]

**Pull request merged**